common/djangoapps/student/views/dashboard.py

```diff
@@ -781,6 +782,22 @@ def student_dashboard(request):
             enr for enr in course_enrollments if entitlement.enrollment_course_run.course_id != enr.course_id
         ]
 
+    course_upgrade_message = ""
+    if request.GET.get('transaction') and request.GET.get('data'):
+        data = request.GET.get('data')
+        decoded_data = data.decode('base64', 'strict')
+        transaction_course, transaction_user = decoded_data.split('#')
+        if transaction_course and transaction_user:
+            log.info(
+                "Approved Authorizenet transaction. Pending course status for Course %s and User %s",
+                transaction_user,
+                transaction_course,
+            )
+            if transaction_user == request.user.username:
+                transaction_course_id = CourseKey.from_string(transaction_course.replace(" ", "+"))
+                course_display_name = CourseOverview.get_from_id(transaction_course_id).display_name
+                course_upgrade_message = "Please wait for the confirmation email as your request is being processed. Course: \"{}\" will be updated soon.".format(course_display_name)
```

Comment:
Problem with this approach is user is going to see this message every time he/she refresh the page no matter we have received the notification from authorize.net? There should be a way to stop showing message once we have notification.

Reply:
Requested changes have been done. Now I have implemented it using cookies.